src/internal/node/node-utf8.js

```diff
@@ -30,9 +29,9 @@ function encode (str) {
 }
 
 function decode (buffer, length) {
-  if (buffer instanceof NodeBuffer) {
+  if (Object.prototype.hasOwnProperty.call(buffer, '_buffer')) {
```

Comment:
Looks like there should be a decode function on NodeBuffer and CombinedBuffer

Reply:
Yes, It should be a method present in all the buffers versions, the issue is way it's done vary a bit between buffer types, I could not figure out a way to do it properly.